### ISSUE TITLE
[java] AssignmentInOperand false positive in 7.17.0 for case blocks in switch statements

### ISSUE THREAD

**Issue by kdandoy107255** [pmd/pmd#6092]
**Affects PMD Version:**
7.17.0
**Rule:**
AssignmentInOperand
Please provide the rule name and a link to the rule documentation:
[AssignmentInOperand](https://docs.pmd-code.org/pmd-doc-7.17.0/pmd_rules_java_errorprone.html#assignmentinoperand)

**Description:**
The fix for issue https://github.com/pmd/pmd/issues/3434 causes version 7.17.0 to flag assignment in case assignments.

**Code Sample demonstrating the issue:**

```java
public final class FPTestAssignmentInOperand {
	
	private FPTestAssignmentInOperand() {}
	public static void main(String[] args)
	{
		String a = null;
		String b = null;
		String c = null;
		
	    switch (args[0])
	    {
	    	case "A" -> a = "a";
	    	case "B" -> b = "b";
	    	case "C" -> c = "c";
	    	default -> {/*Do Nothing*/}
	    }
	    System.out.println("A: " + a + ", B: " + b + ", C: " + c);
	}
}

```

**Expected outcome:**

PMD reports a violation at lines 12,13, and 14, but that's wrong. That's a false positive.

**Running PMD through:** Eclipse

**Comment by Wolf2323** [comment]
I have the same situation, and an similar one, that I guess is the save issue. If not let me know and I open a separate issue:

```java
            final Optional<PlayerConversationState> playerConversationState = ...;
            playerConversationState.ifPresent(conversationState -> activeConversation = conversationState);
```

**Comment by mrclmh** [comment]
@Wolf2323 
I could not reproduce your case. I think your case with lambda expression was fixed in [PMD 7.18](https://github.com/pmd/pmd/releases/tag/pmd_releases%2F7.18.0) with the fix
"https://github.com/pmd/pmd/issues/6075: [java] AssignmentInOperand false positive with lambda expressions".

Could you please retry your example with Version 7.18 and open a new issue, if it still persists?

**Comment by kdandoy107255** [comment]
Just want to clarify that the original issue which I posted still exists in 7.18.
